wsproto/connection.py

```diff
@@ -159,3 +161,35 @@ def events(self):
                     )
         except ParseFailed as exc:
             yield CloseConnection(code=exc.code, reason=str(exc))
+
+
+class H2Wrapper(object):
+    def __init__(self, ws_connection, h2_connection, stream_id):
+        self._ws_connection = ws_connection
+        self._h2_connection = h2_connection
+        self._stream_id = stream_id
+
+    @property
+    def state(self):
+        return self._ws_connection.state
+
+    def send(self, event):
+        data = self._ws_connection.send(event)
+        self._h2_connection.send_data(self._stream_id, data)
+        return self._h2_connection.data_to_send()
+
+    def receive_data(self, data):
+        try:
+            events = self._h2_connection.receive_data(data)
+        except h2.exceptions.ProtocolError:
+            raise RemoteProtocolError("Error in HTTP/2 data")
+        else:
+            for event in events:
+                if isinstance(event, h2.events.DataReceived):
```

Comment:
Do we need to handle any other h2 events? ping-pong?
Raise an error on unexpected events?

Reply:
Your right it should. I've looked at doing so and I don't think wsproto can, instead I think we should limit ourselves to the unmanaged case. The biggest problem is what to do if a window updated event is received. I've updated this pull request with this in mind.

Reply:
Did you just delete basically 95% of this PR - or did the force-push mess something up?

Reply:
I did delete most of this, the approach didn't work.